registry/endpoint.go

```diff
@@ -100,15 +99,18 @@ func newEndpoint(address string, secure bool) (*Endpoint, error) {
 		return nil, err
 	}
 	endpoint.IsSecure = secure
+	tr := NewTransport(ConnectTimeout, endpoint.IsSecure)
+	endpoint.client = HTTPClient(transport.NewTransport(tr, DockerHeaders(metaHeaders)...))
```

Comment:
Ping @stevvooe 

Any endpoint instantiated with `newEndpoint` or `NewEndpoint` is guaranteed to have an instantiated `client` field that can then be used anywhere the endpoint is passed on.

Prior to this, I had an `HTTPClient()` method on `Endpoint` which would lazily instantiate a client without providing `DockerHeaders` which was not the behavior on master. I considered passing `metaHeaders` to the `HTTPClient()` method, but that introduced another problem, which was that the header would be ignored if the client had already be instantiated. If I made it not to ignore the header, then there was a possibility of wrapping the client's existing Transport once more.

The solution I suggest here is to ensure, that each endpoint is instantiated with `metaHeaders` and has a corresponding ready-to-use HTTP client.